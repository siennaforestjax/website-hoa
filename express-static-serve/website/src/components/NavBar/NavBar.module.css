.navHamburger {
  display: block;
  height: 50px;
  width: 50px;
  transform: rotateZ(90deg);
  letter-spacing: -1px;
  top: var(--gutter);
  left: var(--gutter);
  background-color: rgb(var(--primaryButtonColor));
  border: none;
  box-shadow: 0 0 8px -2px black;
  border-radius: var(--buttonBorderRadius);
  position: absolute;
}

.navPanelBackground {
  position: absolute;
  top: 0;
  left: -100vw;
  width: 100vw;
  height: 100vh;
  background-color: transparent;
  z-index: 1;
}

.navPanelBackground.open {
  left: 0;
}

.navPanel {
  --panelWidth: 40vw;
  position: absolute;
  left: calc(-1 * var(--panelWidth));
  width: var(--panelWidth);
  height: auto;
  padding-left: var(--gutter);
  border-radius: 3px;
  transition: none;
  background: linear-gradient(to right, white 5%, rgb(var(--primaryColor)));
  z-index: 2;
}

.navPanel.open {
  top: 0;
  left: 0;
  transition: left 0.3s ease;
  box-shadow: 1px 0 8px 0 black;
}

.navPanel > .exitButton {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  top: 0;
  right: 0;
  padding: 0;
  height: 20px;
  width: 20px;
  position: absolute;
  color: rgb(var(--primaryFontColor));
  transform: rotateZ(45deg);
  background-color: transparent;
  border: none;
}

.navList > li {
  display: block;
  width: 100%;
}

.navList > li {
  font-weight: bold;
  color: rgb(var(--primaryFontColor));
  margin: var(--gutter) 0;
}

.navSubList > a {
  font-weight: normal;
  text-decoration: none;
  display: block;
  width: 90%;
  height: auto;
  padding-left: var(--gutter);
  color: rgb(var(--primaryFontColor));
}

.navSubList > a:hover {
  background-color: white;
}

@media screen and (min-width: 641px) {
  .navPanel {
    --panelWidth: 360px;
    left: calc(-1 * var(--panelWidth));
    width: var(--panelWidth);
  }
}

@media screen and (max-width: 640px) {
  .navPanel {
    --panelWidth: 80vw;
    left: calc(-1 * var(--panelWidth));
    width: var(--panelWidth);
  }

  .navHamburger {
    display: block;
    height: 40px;
    width: 40px;
    transform: rotateZ(90deg);
    letter-spacing: -1px;
    top: calc(1.6 * var(--gutter));
    left: calc(1 * var(--gutter));
    background-color: rgb(var(--primaryButtonColor));
    border: none;
    box-shadow: 0 0 8px -2px black;
    border-radius: var(--buttonBorderRadius);
    position: absolute;
  }
}
