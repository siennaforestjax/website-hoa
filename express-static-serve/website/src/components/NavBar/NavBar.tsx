import React, { useState } from 'react';
import classes from './NavBar.module.css';
import { Link } from 'react-router-dom';

function NavBar() {
  const [isOpen, setIsOpen] = useState(false);

  function toggleNavPanel() {
    setIsOpen(!isOpen);
  }

  return (
    <nav>
      <button className={classes.navHamburger} onClick={() => toggleNavPanel()}>
        l l l
      </button>
      <div className={`${classes.navPanel} ${isOpen ? classes.open : ''}`}>
        <button
          type='button'
          className={classes.exitButton}
          onClick={() => toggleNavPanel()}
        >
          +
        </button>
        <ul className={classes.navList}>
          <li>
            About
            <ul className={classes.navSubList}>
              <Link to='/' onClick={() => toggleNavPanel()}>
                Your Neighborhood (Home)
              </Link>
              <Link to='/board' onClick={() => toggleNavPanel()}>
                Your HOA Board
              </Link>
            </ul>
          </li>
          <li>
            Celebrate
            <ul className={classes.navSubList}>
              <Link to='/yotm' onClick={() => toggleNavPanel()}>
                Yard of the Month
              </Link>
            </ul>
          </li>
          <li>
            HOA Documents
            <ul className={classes.navSubList}>
              <Link to='/documents' onClick={() => toggleNavPanel()}>
                See All
              </Link>
            </ul>
          </li>
          <li>
            Contractor Recommendations
            <ul className={classes.navSubList}>
              <Link to='/recommendations' onClick={() => toggleNavPanel()}>
                See All
              </Link>
            </ul>
          </li>
        </ul>
      </div>
      <div
        className={`${classes.navPanelBackground} ${
          isOpen ? classes.open : ''
        }`}
        onClick={() => toggleNavPanel()}
      ></div>
    </nav>
  );
}

export default NavBar;
