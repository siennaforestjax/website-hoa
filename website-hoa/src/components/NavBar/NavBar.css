.nav-container {
  height: 100%;
  position: relative;
}

.nav-hamburger {
  display: block;
  height: 50px;
  width: 50px;
  transform: rotateZ(90deg);
  letter-spacing: -1px;
  top: var(--gutter);
  left: var(--gutter);
  background-color: var(--primaryButtonColor);
  border: none;
  box-shadow: black 0px 0px 8px -2px;
  border-radius: var(--buttonBorderRadius);
  position: absolute;
}

.nav-panel {
  --panelWidth: 65vw;
  position: absolute;
  width: var(--panelWidth);
  left: calc(-1 * var(--panelWidth));
  transition: none;
  height: auto;
  border-radius: 3px;
}

.nav-panel.open {
  top: 0;
  left: 0;
  transition: left 0.3s ease;
  box-shadow: 1px 0 8px 0 black;
}

.nav-panel > .exit-button {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  top: 0;
  right: 0;
  padding: 0;
  height: 20px;
  width: 20px;
  position: absolute;
  color: var(--primaryFontColor);
  transform: rotateZ(45deg);
  background-color: transparent;
  border: none;
}

.nav-list > li {
  padding-left: var(--gutter);
  font-weight: bold;
}

.nav-list > li,
.nav-sub-list > li {
  display: block;

  line-height: 50px;
  border-bottom: 1px solid black;

  margin: auto;
  text-align: left;
  color: var(--primaryFontColor);
  background-color: var(--secondaryColor);
}

.nav-sub-list > li {
  border-bottom: 1px dashed black;
  line-height: 20px;
  position: relative;
  padding: 5px 5px;
  text-overflow: ellipsis;
  font-weight: normal;
}

.nav-sub-list > li:last-of-type {
  border-bottom: none;
}

.nav-sub-list > li:first-of-type {
  border-top: 1px dashed black;
}

.nav-list > li:last-of-type,
.nav-sub-list > li:last-of-type {
  border-bottom: none;
}

.nav-sub-list > li:hover {
  color: var(--secondaryColor);
  background-color: var(--primaryFontColor);
}
