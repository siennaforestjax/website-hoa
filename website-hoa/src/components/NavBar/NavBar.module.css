.navHamburger {
  display: block;
  height: 50px;
  width: 50px;
  transform: rotateZ(90deg);
  letter-spacing: -1px;
  top: var(--gutter);
  left: var(--gutter);
  background-color: rgb(var(--primaryButtonColor));
  border: none;
  box-shadow: black 0px 0px 8px -2px;
  border-radius: var(--buttonBorderRadius);
  position: absolute;
}

.navPanelBackground {
  position: absolute;
  top: 0;
  left: -100vw;
  width: 100vw;
  height: 100vh;
  background-color: transparent;
  z-index: 1;
}

.navPanelBackground.open {
  left: 0;
}

.navPanel {
  --panelWidth: 65vw;
  position: absolute;
  left: calc(-1 * var(--panelWidth));
  width: var(--panelWidth);
  height: auto;
  border-radius: 3px;
  transition: none;
  background-color: rgb(var(--secondaryColor));
  z-index: 2;
}

.navPanel.open {
  top: 0;
  left: 0;
  transition: left 0.3s ease;
  box-shadow: 1px 0 8px 0 black;
}

.navPanel > .exitButton {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  top: 0;
  right: 0;
  padding: 0;
  height: 20px;
  width: 20px;
  position: absolute;
  color: rgb(var(--primaryFontColor));
  transform: rotateZ(45deg);
  background-color: transparent;
  border: none;
}

.navList a {
  text-decoration: none;
}

.navList > li {
  paddingleft: var(--gutter);
  font-weight: bold;
  border-bottom: 1px solid black;
}

.navList > li,
.navSubList > li {
  display: block;
  line-height: 50px;
  margin: auto;
  text-align: left;
  color: rgb(var(--primaryFontColor));
  background-color: rgb(var(--secondaryColor));
}

.navSubList > li > a {
  display: block;
  width: 100%;
  height: auto;
  padding: 0;
  margin: 0;
  color: rgb(var(--primaryFontColor));
  background-color: rgb(var(--secondaryColor));
}

.navSubList > li {
  border-bottom: 1px dashed black;
  line-height: 20px;
  position: relative;
  padding: 5px 5px;
  text-overflow: ellipsis;
  font-weight: normal;
}

.navSubList > li:last-of-type {
  border-bottom: none;
}

.navSubList > li:first-of-type {
  border-top: 1px dashed black;
}

.navList > li:last-of-type,
.navSubList > li:last-of-type {
  border-bottom: none;
}

.navSubList > li > a:hover {
  color: rgb(var(--secondaryColor));
  background-color: rgb(var(--primaryFontColor));
}
