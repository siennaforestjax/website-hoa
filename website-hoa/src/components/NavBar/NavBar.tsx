import React from 'react';
import './NavBar.css';

const NavBar = ({ ...isLoggedIn }) => {
  function displayContactList() {
    return isLoggedIn ? (
      <li>
        <a>Contact List</a>
      </li>
    ) : (
      <li>Contact List (Log in)</li>
    );
  }

  function toggleNavPanel() {
    document.getElementsByClassName('nav-panel')[0].classList.toggle('open');
  }

  return (
    <nav className="nav-container">
      <button className="nav-hamburger" onClick={toggleNavPanel}>
        l l l
      </button>
      <div className="nav-panel">
        <button type="button" className="exit-button" onClick={toggleNavPanel}>
          +
        </button>
        <ul className="nav-list">
          <li>
            Events
            <ul className="nav-sub-list">
              <li>See All</li>
            </ul>
          </li>
          <li>
            Docs
            <ul className="nav-sub-list">
              <li>Covenants & Restrictions</li>
              {displayContactList()}
            </ul>
          </li>
          <li>
            About
            <ul className="nav-sub-list">
              <li>The Board</li>
              <li>The Neighborhood</li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default NavBar;
