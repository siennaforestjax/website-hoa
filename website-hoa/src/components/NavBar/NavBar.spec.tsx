import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

describe('The NavBar Component', () => {
    describe('The Snapshot', () => {
        it('should render as expected', () => {
            const tree = renderer.create(<NavBar />).toJSON();

            expect(tree).toMatchSnapshot();
        })
    })
})