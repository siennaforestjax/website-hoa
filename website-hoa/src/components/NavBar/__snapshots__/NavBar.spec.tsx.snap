// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NavBar Component The Snapshot should render as expected 1`] = `
<nav
  className="nav-container"
>
  <button
    className="nav-hamburger"
    onClick={[Function]}
  >
    l l l
  </button>
  <div
    className="nav-panel"
  >
    <button
      className="exit-button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <ul
      className="nav-list"
    >
      <li>
        Events
        <ul
          className="nav-sub-list"
        >
          <li>
            See All
          </li>
        </ul>
      </li>
      <li>
        Docs
        <ul
          className="nav-sub-list"
        >
          <li>
            Covenants & Restrictions
          </li>
          <li>
            <a>
              Contact List
            </a>
          </li>
        </ul>
      </li>
      <li>
        About
        <ul
          className="nav-sub-list"
        >
          <li>
            The Board
          </li>
          <li>
            The Neighborhood
          </li>
        </ul>
      </li>
    </ul>
  </div>
</nav>
`;
